R/AllGenerics.R

```diff
@@ -12,10 +12,15 @@ setGeneric("backendMerge", def = function(object, ...)
     standardGeneric("backendMerge"),
     valueClass = "MsBackend")
 #' @rdname hidden_aliases
-setGeneric("bin", function(object, ...) standardGeneric("bin"))
+setGeneric("bin", function(x, ...) standardGeneric("bin"))
+#' @rdname hidden_aliases
+setMethod("bin", "numeric", MsCoreUtils::bin)
```

Comment:
Just thinking out loud here. Could it be that the following will be faster?
```r
.bin <- MsCoreUtils::bin
setMethod("bin", "numeric", .bin)
```

If you remember our benchmarks on slack:
```r
x <- 1:1e4
f1 <- function(x) { for(i in x)base::sort(x) }
f2 <- function(x) { for(i in x)sort(x) }
f3 <- function(x) { .sort <- base::sort; for(i in x).sort(x) }
microbenchmark(always=f1(x), never=f2(x), once=f3(x))
# Unit: milliseconds
#    expr   min    lq  mean median     uq    max neval
#  always 77.69 81.93 91.54  84.12 101.91 151.08   100
#   never 35.10 37.12 40.89  38.25  42.84  61.31   100
#    once 35.36 37.38 41.58  38.40  44.17  68.43   100
```

Reply:
Don't struggle with this. I will try it myself (because the dispatching happens in `.peaks_bin` actually) once the PR is merged. So ignore it for now.

Reply:
OK, but yes, that makes sense.